interpreter/src/main/scala/pl/touk/nussknacker/engine/compile/nodecompilation/FragmentParameterValidator.scala

```diff
@@ -0,0 +1,76 @@
+package pl.touk.nussknacker.engine.compile.nodecompilation
+
+import pl.touk.nussknacker.engine.api.NodeId
+import pl.touk.nussknacker.engine.api.context.ProcessCompilationError.{
+  InitialValueNotPresentInPossibleValues,
+  InvalidParameterInputConfig,
+  RequireValueFromEmptyFixedList
+}
+import pl.touk.nussknacker.engine.api.context.{ProcessCompilationError, ValidationContext}
+import pl.touk.nussknacker.engine.graph.expression.Expression
+import pl.touk.nussknacker.engine.graph.node.FragmentInputDefinition.{
+  FixedExpressionValue,
+  FragmentParameter,
+  FragmentParameterInputMode
+}
+
+object FragmentParameterValidator {
+
+  def validate(
+      fragmentParameter: FragmentParameter,
+      fragmentInputId: String,
+      compiler: NodeCompiler,
+      validationContext: ValidationContext // localVariables must include this and other FragmentParameters
+  )(implicit nodeId: NodeId): List[ProcessCompilationError] = {
+    val inputConfigResponse = if (fragmentParameter.inputConfig.isValid) {
+      List.empty
+    } else {
+      List(InvalidParameterInputConfig(fragmentParameter.name, Set(fragmentInputId)))
```

Comment:
this could provide more telling errors, but it's only possible if the request is malformed, so idk